rp2040-hal/src/pio.rs

```diff
@@ -519,10 +519,20 @@ impl<SM: ValidStateMachine, State> StateMachine<SM, State> {
         self.sm.sm().sm_addr.read().bits()
     }
 
-    /// Set the current instruction.
+    #[deprecated(note = "Renamed to exec_instruction")]
+    ///Execute the instruction immediately.
     pub fn set_instruction(&mut self, instruction: u16) {
-        // TODO: Check if this function is safe to call while the state machine is running.
-        self.sm.set_instruction(instruction);
+        self.exec_instruction(instruction);
+    }
+
+    ///Execute the instruction immediately.
+    pub fn exec_instruction(&mut self, instruction: u16) {
+        // This is allowed even if the state machine is running.
+        //
+        // However, the datasheet says:
+        // "If EXEC instructions are used, instructions written to INSTR must not stall"
+        // It's unclear what happens if this is violated.
+        self.sm.exec_instruction(instruction);
```

Comment:
I think this warning should be in the function's documentation.
```suggestion
    /// Execute the instruction immediately.
    /// 
    /// While this is allowed even when the state machine is running, the datasheet says:
    /// > If EXEC instructions are used, instructions written to INSTR must not stall.
    /// It's unclear what happens if this is violated.
    pub fn exec_instruction(&mut self, instruction: u16) {
        // TODO: clarify what happens if the instruction stalls.
        self.sm.exec_instruction(instruction);
```